src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -298,13 +306,15 @@ private static BoundExpression CreateAnonymousFunctionConversion(SyntaxNode synt
             // UNDONE: Figure out what to do about the error case, where a lambda
             // UNDONE: is converted to a delegate that does not match. What to surface then?
 
-            var unboundLambda = (UnboundLambda)source;
+            Debug.Assert(source.KindIgnoringSuppressions() == BoundKind.UnboundLambda);
+            var unboundLambda = (UnboundLambda)source.RemoveSuppressions();
```

Comment:
This is an example of the type of change that worries me a bit. I feel like there are lots of places in the code that are now going to need similar changes. On the other hand if suppression was handled by simply rewriting the node on which suppression targeted with a new type + a flag to indicate "yes this is the result of a suppression" then most of existing code would continue to work as is.

Reply:
Indeed, this was a lot of hunting. Tagging @gafter for thoughts.